### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by Krishnansh5**
- sending data in batches as described in frontend PR
- revoked APC access for inactive RCs

### CODE REVIEW

**Review by abhishekshree — DISMISSED** [review]
lgtm

**Comment by abhishekshree** [comment]
@Krishnansh5 document the rc/util.admin.go before merging

File: company/admin.get_company.go [hunk 1]
```diff
@@ -38,3 +38,28 @@ func getCompanyHandler(ctx *gin.Context) {
 
 	ctx.JSON(http.StatusOK, company)
 }
+
+func getLimitedCompaniesHandler(ctx *gin.Context) {
+	var companies []Company
+
+	pageSize := ctx.DefaultQuery("pageSize", "100")
+	lastFetchedId := ctx.Query("lastFetchedId")
+	pageSizeInt, err := strconv.ParseInt(pageSize, 10, 32)
```
Comment (1-Harshit) [inline company/admin.get_company.go:47]: Don't we have util func for this?

File: company/admin.get_company.go [hunk 2]
```diff
@@ -38,3 +38,28 @@ func getCompanyHandler(ctx *gin.Context) {
...
+	if err != nil {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	lastFetchedIdInt, err := strconv.ParseUint(lastFetchedId, 10, 32)
```
Comment (1-Harshit) [inline company/admin.get_company.go:52]: Again, util func?

File: company/router.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ func AdminRouter(r *gin.Engine) {
 	{
 		admin.GET("", getAllCompaniesHandler)
 		admin.GET("/:cid", getCompanyHandler)
+		admin.GET("/limited/:pageSize/:lastFetchedId", getLimitedCompaniesHandler)
```
Comment (1-Harshit) [inline company/router.go:13]: I think we can have pagesize and lastpage page as query params instead of path params

File: rc/util.admin.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package rc
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+func ensurePseudoAdminAccess() gin.HandlerFunc {
```
Comment (1-Harshit) [inline rc/util.admin.go:9]: Can you change file name to display it's a middleware? Also, ensurepsuedoadmin is already a function can you changed to something else? Like checkRCStatus? Or ensureActiveRC?

File: student/router.go [hunk 5]
```diff
@@ -18,6 +18,7 @@ func AdminRouter(r *gin.Engine) {
 	{
 		admin.DELETE("/:sid", deleteStudentHandler)
 		admin.GET("", getAllStudentsHandler)
+		admin.GET("/limited", getLimitedStudentsHandler)
```
Comment (1-Harshit) [inline student/router.go:21]: Limited? Pakka? 😂

**Review by 1-Harshit — CHANGES_REQUESTED** [review]
Good PR, suggested some changes

**Comment by abhishekshree** [comment]
@1-Harshit full nitpicking i see i see

**Comment by 1-Harshit** [comment]
Left some 'suggestions' is all :)
Won't do if you're so against it.

**Comment by abhishekshree** [comment]
Nahi bhai teri bhi repo hai

**Comment by utkarsh-tf141** [comment]
-closes #100 #102

**Pull request merged**